crates/starknet/src/starknet/mod.rs

```diff
@@ -231,6 +336,10 @@ impl Starknet {
         // clear pending block information
         self.generate_pending_block()?;
 
+        if self.config.dump_on.is_some() && self.config.dump_on == Some(DumpMode::OnTransaction) {
```

Comment:
```suggestion
        if self.config.dump_on == Some(DumpMode::OnTransaction) {
```

Reply:
Fixed in 2e2312e9305304bcc68c0c2858ff094e6edf5a91